### PR TITLE
solution

### PR DESCRIPTION
**Pull request by GoldenSpade**

### CODE REVIEW

**Review by leonid-vegera — APPROVED** [review]
Good job!

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,24 @@
...
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    if (this.length === 0) {
+      return string;
+    }
```
Comment (varseniuk) [inline src/arrayMethodJoin.js:12]:
```suggestion
```
This check is redundant. If this.length === 0 then the loop will not even start and in the end, the function will still return `string`.
'string`, by the way, is a bad name for the variable. Use something more meaningful in the future

**Review by varseniuk — APPROVED** [review]
Good job! 👍 
Left one comment

File: src/arrayMethodJoin.js (hunk 1 above)
Comment (BilliZane) [reply src/arrayMethodJoin.js:12]: Ok

**Pull request closed without merging**